prime-router/src/main/kotlin/azure/Event.kt

```diff
@@ -96,80 +98,52 @@ abstract class Event(val eventAction: EventAction, val at: OffsetDateTime?) {
         fun parseQueueMessage(event: String): Event {
```

Comment:
How are we planning to handle the deployment scenario where the poison queue has messages of the old custom string format? Those messages would no longer be able to be processed, right? We either need to:

1. Leave the string processing code to handle the old poison queue scenario then deploy then create new PR to remove the string processing code once the poison queues no longer contain messages of the old format. If we do it this way, we will need tests for both cases (new and old). Also, we should create the PR to remove the code in advance so we can just merge it whenever the queues empty.
2. Before merging this PR to staging, make sure all poison queues are empty. Then, before deploying to prod, make sure all poison queues are empty. I would probably prefer this solution because it is less code changes (minimizes possibility of introducing defects) but may only be doable if the queues are empty or close to empty (check the state of them in azure).

Whichever strategy you decide on, it should be documented in the Pull Request.

Also this issue is called out in the risks section of the ticket.